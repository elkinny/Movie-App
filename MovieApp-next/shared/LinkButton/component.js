import React from 'react';
import Link from 'next/link'
import PropTypes from 'prop-types';

import styles from './style.scss';

const LinkButton = props => {
  const { to, asValue, cssClass, children } = props;
  return (
    <Link href={to} as={asValue || 'url'} className={cssClass}>
      <span className={styles.link}>{children}</span>
    </Link>
  );
};

export default LinkButton;

LinkButton.propTypes = {
  to: PropTypes.string.isRequired,
  cssClass: PropTypes.string,
  as: PropTypes.string,
};
