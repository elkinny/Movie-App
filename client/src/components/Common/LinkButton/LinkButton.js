import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import styles from './link-button.scss';

const LinkButton = props => {
  return (
    <Link to={props.to}>
      <span className={styles.link}>{props.children}</span>
    </Link>
  );
};

export default LinkButton;

LinkButton.propTypes = {
  to: PropTypes.string.isRequired,
};
