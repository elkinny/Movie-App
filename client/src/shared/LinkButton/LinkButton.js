import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import styles from './link-button.scss';

const LinkButton = props => {
  const { to, cssClass, children } = props;
  return (
    <Link to={to} className={cssClass}>
      <span className={styles.link}>{children}</span>
    </Link>
  );
};

export default LinkButton;

LinkButton.propTypes = {
  to: PropTypes.string.isRequired,
  cssClass: PropTypes.string,
};
