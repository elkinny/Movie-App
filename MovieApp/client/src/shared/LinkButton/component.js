import React from 'react';
import { Link } from 'react-router-dom';

import { StyledLinkButton } from './styled.component';

//@flow

type Props = {
  to: ?string,
  cssClass: string,
};

const LinkButton = (props: Props) => {
  const { to, cssClass, children } = props;
  return (
    <Link to={to} className={cssClass}>
      <StyledLinkButton>{children}</StyledLinkButton>
    </Link>
  );
};

export default LinkButton;
