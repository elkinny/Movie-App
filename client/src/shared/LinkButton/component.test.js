import LinkButton from './LinkButton';
import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';

describe('LinkButton: ', () => {
  it('LinkButton to match snapshot', () => {
    const props = {
      to: '/',
    };
    const component = mount(
      <BrowserRouter>
        <LinkButton {...props} />
      </BrowserRouter>,
    );
    expect(component).toMatchSnapshot();
  });
});
