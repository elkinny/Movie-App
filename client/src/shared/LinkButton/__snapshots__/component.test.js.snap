// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton:  LinkButton to match snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <LinkButton
      to="/"
    >
      <Link
        replace={false}
        to="/"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <span
            className="link"
          />
        </a>
      </Link>
    </LinkButton>
  </Router>
</BrowserRouter>
`;
